src/pages/test-results-search/test-results-search.ts

```diff
@@ -34,6 +35,7 @@ export class TestResultsSearchPage extends BasePageComponent {
   hasSearched: boolean = false;
   showSearchSpinner: boolean = false;
   showAdvancedSearchSpinner: boolean = false;
+  subscription: Subscription;
```

Comment:
We can do multiple types searches , without leaving the page. 

The unsubscribe will only work on the last search you did , leaving the others orphaned. Should subscription be an array we push all the searches to and loop through and unsubscribe at the end.

Reply:
Thanks Tom,

I've updated the search functions on this to unsubscribe from any previous subscriptions when they are called as well as unsubscribe from the last subscription when leaving the page. Initially setting the subscription to Subscription.Empty means there is no need to check if we have already subscribed to something.